### PR TITLE
Add support for hidden recipients during decryption.

### LINKED ISSUE #1275: Support decryption with hidden recipients.
RNP reports that it cannot decrypt if no Key ID of the Public-Key Encrypted Session Key Packets matches a known private key.

It would be great if it instead could check for "hidden" recipients. In case the Key ID `0x00000000` is present, RNP should try to decrypt with all known private keys. See <https://tools.ietf.org/html/rfc4880#section-5.1> (last paragraph of that section).

Background: Currently Thunderbird 78 an later fails to decrypt messages where the user is a hidden recipient. Such ciphertexts could e.g. be created by Enigmail, or in general with GnuPG's `hidden-encrypt-to` config option.  Downstream issue: <https://thunderbird.topicbox.com/groups/e2ee/T7e31aaa38399d4b3-Mefc2734288f0dfa17d01b823/fails-to-decrypt-with-recipient-key-id-0x00000000-hidden-recipient>.

### PR DESCRIPTION
**Pull request by ni4**
This includes quite a bunch of changes, so took long time to implement.
Please review carefully.

Fixes #1275

### CODE REVIEW

**Review by ronaldtse — APPROVED** [review]
Thank you @ni4 !

File: src/rnp/rnp.1.adoc [hunk 1]
```diff
@@ -332,6 +332,12 @@ Override or set a file name, stored inside of OpenPGP message. +
 By default RNP will store input filename (or empty string for *stdin*/*env* input) in the resulting OpenPGP message during encryption or embedded signing.
 This option allows to override this. Special value *_CONSOLE* may be used for "for your eyes only"-message. Refer OpenPGP documentation for the details.
 
+*--allow-hidden* ::
+Allow hidden recipient support. +
++
+Senders of the encrypted message may wish to hide the recipient by replacing his keyid, stored inside OpenPGP message, with zeroes.
+This could require to try every secret key against such message in order to be able to decrypt, and is disabled by default.
+
```
Comment (antonsviridenko) [inline src/rnp/rnp.1.adoc:340]: Need some better wording like 
"Sender of an encrypted message may wish to hide recipient's key by setting a Key ID field to all zeroes. In this case receiver has to try every available secret key, checking for a valid decrypted session key. This option is disabled by default."

Also it's not clear from this description if this option applies to sender or receiver.

Comment (ni4) [reply src/rnp/rnp.1.adoc:340]: @antonsviridenko Thanks, updated the PR.

**Review by antonsviridenko — APPROVED** [review]

**Comment by ni4** [comment]
@ronaldtse @antonsviridenko Should we merge this now?

**Comment by ronaldtse** [comment]
Thank you @ni4! I believe we were just waiting for the tests to pass before.

**Pull request merged**